### PR TITLE
Jenkins: more refactoring

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: tools/jenkins/util.groovy [hunk 1]
```diff
@@ -160,7 +169,7 @@ def slack(build, env) {
     }
 }
 
-def cmake(opts, externalModules, onModules, offModules) {
+def cmake(Map opts, List externalModules, List onModules, List offModules) {
```
Comment (ResearchDaniel) [inline tools/jenkins/util.groovy]: externalModulePaths

File: tools/jenkins/util.groovy [hunk 2]
```diff
@@ -219,14 +233,16 @@ def build(opts, externalModules, onModules, offModules = []) {
     }    
 }
 
-def buildStandard(params, externalModules, extraOpts, onModules, offModules = []) {
+// Args: params, external, opts, onModules, offModules
+def buildStandard(Map args = [:]) {
+    assert args.params, "Arguemnt params must be supplied"
```
Comment (ResearchDaniel) [inline tools/jenkins/util.groovy]: Typo

File: tools/jenkins/util.groovy [hunk 3]
```diff
@@ -176,11 +185,10 @@ def clean(params) {
     }
 }
 
-Map defaultOptions(params) {
+Map defaultOptions(String buildType) {
```
Comment (ResearchDaniel) [inline tools/jenkins/util.groovy]: -> defaultCMakeOptions

File: Jenkinsfile [hunk 4]
```diff
@@ -14,9 +14,9 @@ node {
     try {
         def external = []
         def extraOn = []
-        def off = ["ABUFFERGL" , "DISCRETEDATA", "GLFW", "HDF5"]
+        def off = ["ABUFFERGL" , "DISCRETEDATA", "HDF5"]
 
-        util.buildStandard(params, external, extraOn, off)
+        util.buildStandard(params : params, external : external, onModules : extraOn,  offModules : off)
```
Comment (ResearchDaniel) [inline Jenkinsfile]: externalModulePaths?

**Review by ResearchDaniel — APPROVED** [review]
Looks good. Suggested some make changes to clarify some commands

File: tools/jenkins/util.groovy [hunk 5]
```diff
@@ -93,15 +105,16 @@ def integrationtest(display = 0) {
...
     cmd('Regression Tests', 'regress', ['DISPLAY=:' + display]) {
+        def modules = modulepaths ? '--modules ' + modulepaths.join(' ') : ''
         try {
             sh """
                 python3 ../inviwo/tools/regression.py \
                         --inviwo ../build/bin/inviwo \
                         --header ${env.JENKINS_HOME}/inviwo-config/header.html \
                         --output . \
-                        --repos ../inviwo
+                        --repos ../inviwo ${modules}
```
Comment (ResearchDaniel) [inline tools/jenkins/util.groovy]: A question that applies to some of the other tests (copyright) as well:
Should inviwo directory always be included?
External modules only want to test their own repos, not inviwo (which should already be tested anyway)

Comment (petersteneteg) [reply tools/jenkins/util.groovy]: true

**Comment by petersteneteg** [comment]
tested by jenkins

**Comment by petersteneteg** [comment]
tested by jenkins

**Comment by petersteneteg** [comment]
tested by jenkins

**Comment by petersteneteg** [comment]
tested by jenkins again

**Pull request merged**